src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * @author leslieharland
+ */
+
+package duke.task;
+
+import duke.command.Operation;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Class Task.
+ */
+public abstract class Task implements TaskPrintable, Parseable {
+
+    protected String description;
+
+    protected boolean isDone;
+
+    protected String type;
+
+    protected String userKeyedDateString;
+
```

Comment:
It would be better if there were no newlines between each class member, so it is easier to understand.